arch/arm/core/cortex_m/mpu/arm_mpu_v8_internal.h

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2017 Linaro Limited.
+ * Copyright (c) 2018 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <cmse.h>
+
+/* Global MPU configuration at system initialization. */
+static void _mpu_init(void)
+{
+	/* Configure the cache-ability attributes for all the
+	 * different types of memory regions.
+	 */
+
+	/* Flash region(s): Attribute-0
+	 * SRAM region(s): Attribute-1
+	 */
+	MPU->MAIR0 =
+		((MPU_MAIR_ATTR_FLASH & MPU_MAIR0_Attr0_Msk) >>
+			MPU_MAIR0_Attr0_Pos)
+		|
+		((MPU_MAIR_ATTR_SRAM & MPU_MAIR0_Attr1_Msk) >>
+			MPU_MAIR0_Attr1_Pos);
+}
+
+/* This internal function performs MPU region initialization.
+ *
+ * Note:
+ *   The caller must provide a valid region index.
+ */
+static void _region_init(u32_t index, struct arm_mpu_region *region_conf)
+{
+	ARM_MPU_SetRegion(
```

Comment:
Here, I used, directly the ARM CMSIS provided function to set up the MPU region.